char/char.mbt

```diff
@@ -80,3 +80,33 @@ test "debug_write" {
 pub fn hash(self : Char) -> Int {
   self.to_int()
 }
+
+fn compare_case_insensitive(self: Char, b: Char) -> Int {
+  fn is_ascii_letter(c: Char) -> Bool {
+    ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z')
+  }
+  fn to_uppercase(c: Char) -> Char {
+    // NOTE: 'a' - 'A' == 0x20
+    if 'a' <= c && c <= 'z' { Char::from_int(c.to_int() - 0x20) } else { c }
+  }
+  if is_ascii_letter(self) && is_ascii_letter(b) { 
+    to_uppercase(self).compare(to_uppercase(b)) 
+  } else {
+    self.compare(b)
+  }
```

Comment:
```suggestion
  to_uppercase(self).compare(to_uppercase(b)) 
```
There's too many checks here

Reply:
@peter-jerry-ye  It's a trap here🤣, it causes this case to fail: 

`'_'(0x5F) < 'n'(0x6E) == false` because `'_'(0x5F) > 'N'(0x4E)`. 

Convert to uppercase before comparing only applies to both chars are ASCII letters. This has already been covered in test cases.

Reply:
Then your implementation is incorrect as:
- 'N' < '_' < 'n'
- 'N' == 'n'

It is no longer a totally ordered set

Reply:
Yes, it's theologically a partial order set, likes the float set with NaN. But the results of comparison is guaranteed to be deterministic.

Reply:
Sorry, I mean it is no longer a strict partial order set